net/ipv4/tcp_output.c

```diff
@@ -2794,13 +2767,14 @@ static bool tcp_write_xmit(struct sock *sk, unsigned int mss_now, int nonagle,
 							  max_segs),
 						    nonagle);
 #ifdef CONFIG_SECURITY_TEMPESTA
-		result = tcp_tfw_sk_prepare_xmit(sk, skb, mss_now, &limit,
-						 &nskbs);
-		if (unlikely(result)) {
-			if (result == -ENOMEM)
-				break; /* try again next time */
-			tcp_tfw_handle_error(sk, result);
-			return false;
+		if (sk->sk_write_xmit && skb_tfw_tls_type(skb)) {
+			if (unlikely(limit <= TLS_MAX_OVERHEAD)) {
+			    net_warn_ratelimited("%s: too small MSS %u"
+						 " for TLS\n",
+						 __func__, mss_now);
+				break;
+			}
+			TFW_ADJUST_TLS_OVERHEAD(limit);
```

Comment:
Because of it I adjust it in Tempesta FW code, and push less count of skb. For example if our send window is 3 i push only one skb from Tempesta.  Becuase we can also increase count of skb when we make tls record